webapp/lib/view.dart

```diff
@@ -886,53 +850,29 @@ class TagListView extends BaseView {
     _tagsContainer = new DivElement()
       ..classes.add('tags');
     _tagsActionContainer = new SpanElement()
-      ..classes.add('tags__actions')
-      ..append(
-        new ButtonElement()
-          ..classes.add('button-add-tag')
-          ..text = '+'
-          ..onClick.listen((event) {
-            event.stopPropagation();
-            if (tagsEditable) {
-              var tagElement = new TagView('', model.TagType.Normal, onTagChangedCallback, tagsEditable);
-              _tagsActionContainer.insertAdjacentElement('beforebegin', tagElement.renderElement);
-              tagElement.focus();
-              return;
-            }
-            _addTagDropdown(availableTags, onTagChangedCallback);
-          })
-      );
+      ..classes.add('tags__actions');
+    _tagsActionContainer.append(
+      new AddButtonElement((MouseEvent event) {
+        if (tagsEditable) {
+          var editableTag = new EditableTagElement(onTagChangedCallback);
+          _tagsActionContainer.insertAdjacentElement('beforebegin',editableTag.renderElement);
+          editableTag.focus();
+        } else {
+          _tagsActionContainer.append(new DropdownElement(availableTags.keys.toList(), (MouseEvent event) {
```

Comment:
can you extract `DropdownElement` into a local variable? it's pretty hard to read this code, esp. with with the callback function and then `.renderElement`.

Reply:
Simplified it